csrc/scheduler/reduction.cpp

```diff
@@ -1593,17 +1593,27 @@ void scheduleReduction(Fusion* fusion, const ReductionParams* rparams) {
 
   scheduler_utils::moveNonConcretizedBroadcastInnermost(fusion, {reference_tv});
 
-  reduction_scheduler_utils::multiReductionInliner(
-      fusion,
+  // Propagate transformations before we rfactor the other reductions
+  auto reduction_tv = reduction_tvs.at(0);
+  propagateTransformation(reference_tv);
+  // If reduction_tv is rfactored, rfactor all reductions.
+  if (reference_tv != reduction_tv) {
+    propagateRFactor(reference_tv, reduction_tv, reduction_tvs);
+  }
+
+  const auto& unroll_vectorizable_cached_tvs = getCachedTvsToUnrollOrVectorize(
+      reference_tv, is_vectorize, cached_inputs, cached_outputs);
+
+  reduction_scheduler_utils::propagateParallelization(
       reduction_tv,
       reference_tv,
       unroll,
-      vectorize,
       use_iter_grouped_reduction,
-      rparams->unroll_factor_inner_reduction,
       reduction_tvs,
-      cached_inputs,
-      cached_outputs);
+      unroll_vectorizable_cached_tvs);
+
```

Comment:
In reduction scheduler, no need to call `sharedMemoryConsumerVectorization` and `Remove dummy outputs` since these two functions are for persistent buffers.

Reply:
I see the old function call just skipped those args using the default. Thanks for the comment here.

```
    std::vector<TensorView*> smem_persistent_buffer_consumers = {},
    std::vector<TensorView*> dummy_outputs = {});
```